protocol/protocol_config.go

```diff
@@ -0,0 +1,63 @@
+package protocol
+
+import (
+	"github.com/s-rah/onionscan/report"
+	"strings"
+)
+
+type ProtocolConfig struct {
+	TorProxyAddress string
+	DirectoryDepth  int
+}
+
+func Configure(torProxyAddress string, directoryDepth int) *ProtocolConfig {
+	onionScan := new(ProtocolConfig)
+	onionScan.TorProxyAddress = torProxyAddress
+	onionScan.DirectoryDepth = directoryDepth
+	return onionScan
+}
+
+func (os *ProtocolConfig) Scan(hiddenService string) (*report.OnionScanReport, error) {
```

Comment:
I'm torn on this. It makes sense from a conciseness viewpoint to have the Scan function in this file - although it feels weird to have the Scan function operate from the ProtocolConfig type. What do you think about moving the Config to onionscan/config/onionscan_config.go and keep onionscan.go with the Scan function (and pass in an OnionScanConfig ?)